modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -1010,6 +1079,30 @@ public List<String> assignments() {
                 .collect(Collectors.toList());
     }
 
+    /** {@inheritDoc} */
+    // TODO: IGNITE-17256 Use a placement driver for getting a primary replica.
+    @Override
+    public List<IgniteBiTuple<String, Long>> leaderAssignmentsWithTerm() {
+        List<Entry<RaftGroupService>> entries = new ArrayList<>(partitionMap.int2ObjectEntrySet());
+        List<CompletableFuture<LeaderWithTerm>> futs = new ArrayList<>();
+
+        entries.sort(Comparator.comparingInt(Entry::getIntKey));
```

Comment:
I see a few places in the class where we are sorting the map. Let's rewrite it to use an already sorted map - Int2ObjectSortedMap

Reply:
Created a separate ticket for this improvement https://issues.apache.org/jira/browse/IGNITE-18691